src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1759,52 +1760,45 @@ internal virtual void AggregateRenderPackages(IEnumerable<HelixRenderPackage> pa
-                            // edges of meshes. Draw them with a different thickness.
-                            lineGeometry3D = CreateLineGeometryModel3D(rp, rp.MeshVertices.Any()?0.5:1.0);
-                            Element3DDictionary.Add(id, lineGeometry3D);
-                        }
-
-                        var lineSet = lineGeometry3D.Geometry == null ? HelixRenderPackage.InitLineGeometry()
-                        : lineGeometry3D.Geometry as LineGeometry3D;
-                        var startIdx = lineSet.Positions.Count;
-
-                        lineSet.Positions.AddRange(l.Positions);
-                        if (drawDead)
-                        {
-                            lineSet.Colors.AddRange(Enumerable.Repeat(defaultDeadColor, l.Positions.Count));
-                        }
-                        else
+                        //If we are using IRenderInstances data then we need to create a unique Geometry3D object for each instancable item and add instance transforms.
+                        //If we any mesh geometry was not associated with am instance, remove the previously added line data from the render package so the remaining lines can be added to the scene.
+                        if (rp.LineVerticesRangesAssociatedWithInstancing.Any())
                         {
-                            lineSet.Colors.AddRange(l.Colors.Any()
-                             ? l.Colors
-                             : Enumerable.Repeat(defaultLineColor, l.Positions.Count));
+                            //For each range of mesh vertices add the mesh data and instances to the scene
+                            var lineVertexCountTotal = 0;
+                            var j = 0;
+                            foreach (var item in rp.LineVerticesRangesAssociatedWithInstancing)
+                            {
+                                var range = item.Value;
+                                var index = range.Item1; //Start line vertex index
+                                var count = range.Item2 - range.Item1 + 1; //Count of line vertices
+                                var uniqueId = baseId + ":" + j + LinesKey;
+
+                                List<Matrix> instances;
+                                if (rp.instances.TryGetValue(item.Key, out instances))
+                                {
+                                    AddLineData(uniqueId, rp, index, count, drawDead, baseId, instances);
+                                }
+
+                                //Track cumulative total of line vertices added.
+                                lineVertexCountTotal += count;
+                                j++;
+                            }
+
+                            //If all the line regions had texture map data then we are done with line data.
+                            if (lineVertexCountTotal == l.Positions.Count  & !rp.Mesh.Positions.Any())
```

Comment:
bitwise `&`?

Reply:
Can you explain what you're checking for here?

Reply:
this appears incorrect, line vs mesh and should be binary &&